src/main/java/org/zaproxy/admin/GenerateAddonsYAML.java

```diff
@@ -0,0 +1,108 @@
+ * limitations under the License.
+ */
+package org.zaproxy.admin;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.nio.file.Files;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Command line tool for generating the addons markdown file for website.
+ *
+ * @author nirojan
+ */
+public class GenerateAddonsYAML {
+
+    private static final String ZAP_VERSIONS_FILE_NAME = "ZapVersions-2.9.xml";
+    private static final String NEXT_LINE = "\n";
+    private static final String EMPTY_STRING = "";
+    private static final String OUTPUT_DIR = "addons.yml";
+
+
+    public static void main(String[] args) throws Exception {
+
+        StringBuilder sb = new StringBuilder("---" + NEXT_LINE);
+        File xmlFile = new File(ZAP_VERSIONS_FILE_NAME);
+        if(xmlFile.exists()){
+
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            org.w3c.dom.Document doc = builder.parse(ZAP_VERSIONS_FILE_NAME);
+
+            NodeList list = doc.getElementsByTagName("addon");
+
+            for (int i=0; i<list.getLength(); i++) {
+
+                Element addonParentElement = (Element)list.item(i);
+                NodeList addonIndividualNode = doc.getElementsByTagName("addon_" + addonParentElement.getTextContent());
+
+                if(addonIndividualNode.getLength() != 0){
+                    Element addonInfo = (Element)addonIndividualNode.item(0);
+
+                    String addonName = getAddonDetails("name", addonInfo);
+                    String description = getAddonDetails("description", addonInfo);
+                    String author = getAddonDetails("author", addonInfo);
+                    String version = getAddonDetails("version", addonInfo);
+                    String file = getAddonDetails("file", addonInfo);
+                    String status = getAddonDetails("status", addonInfo);
+                    String url = getAddonDetails("url", addonInfo);
+                    String date = getAddonDetails("date", addonInfo);
+                    String size = getAddonDetails("size", addonInfo);
+
+                    sb.append("- name: " + addonName + NEXT_LINE)
+                            .append("  description: " + description + NEXT_LINE)
+                            .append("  author: " + author + NEXT_LINE)
+                            .append("  version: " + version + NEXT_LINE)
+                            .append("  file: " + file + NEXT_LINE)
+                            .append("  status: " + status + NEXT_LINE)
+                            .append("  url: " + url + NEXT_LINE)
+                            .append("  date: " + date + NEXT_LINE)
+                            .append("  link: " + EMPTY_STRING + NEXT_LINE)
```

Comment:
Worth renaming to `infourl` to be more clear. Also, worth stripping the site domain (if the same) to allow local tests/checks.

Reply:
The site should also be a property of the task.

Reply:
Regarding these additions/renames, worth checking zaproxy/zaproxy-website#18 for exact location/case of the fields.